app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
Well, that just means it doesn't reproduce for me. I had only one reproduction and then everything worked fine every time.

Is it fine if I do parenthesis:
```ruby
-    return nil unless target.respond_to? :notification_type || recipient != actor
+    return nil unless (target.respond_to?(:notification_type) || recipient != actor)
```